### PR TITLE
Change reminder and notification scheduling and add summary email

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

Estimated land date reminders are now being sent on the first day of the month.
If number of reminders reaches a threshold and a user has enabled email reminders, then a summary email will be sent instead of individual emails.

The task that sends individual subscriptions has been removed. The views related to individual subscriptions will be removed later.

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

**Review by tomkingston — APPROVED** [review]
Looks like you've manage to deal with a lot of difficulties around sending monthly summaries! 
I notice that this will no longer work for an arbitrary number of days (eg 45 / 100 or whatever), but I guess we only need to worry about 30 or 60 for the estimated land date (although no recent interaction will need to allow any number).
I wonder whether the way this works now means we should be changing the UI a bit - the use of reminder days seems a bit strange now that email reminders sometimes are sent on a monthly basis.

File: datahub/reminder/utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from dateutil.relativedelta import relativedelta
+
+
+REMINDER_DAYS_MAPPING = {
+    30: lambda day: day + relativedelta(months=1),
+    60: lambda day: day + relativedelta(months=2),
+}
+
+
+def reminder_days_to_date_filter(current_date, reminder_days):
```
[the remark this answers is not in the document]
Comment (elcct) [reply datahub/reminder/utils.py:10]: 1. This type of notification will not use arbitrary days and it was decided that 30 and 60 days are just labels for 1 and 2 months. 
2. I think this is out of scope, the user will not be able to enter other values than 30 and 60 for this notification.

File: datahub/reminder/tasks.py [hunk 2]
```diff
@@ -63,26 +68,42 @@ def generate_estimated_land_date_reminders_for_subscription(subscription, curren
...
+    if summary_threshold and subscription.email_reminders_enabled:
+        send_estimated_land_date_summary(
+            projects=list(projects),
+            adviser=subscription.adviser,
+            current_date=current_date,
+        )
+
+    for project in projects:
+        day_diff = (project.estimated_land_date - current_date).days
+        days_left = 30 if day_diff < 32 else 60
```
[the remark this answers is not in the document]
Comment (elcct) [reply datahub/reminder/tasks.py:98]: Correct

File: datahub/investment/project/notification/emails.py [hunk 3]
```diff
@@ -18,17 +21,75 @@ def send_estimated_land_date_reminder(project, adviser, days_left):
...
+        'estimated_land_date': project.estimated_land_date.strftime('%-d %B %Y'),
+    }
+
+
+def get_projects_summary_list(projects):
+    """Gets formatted projects summary list."""
+    notifications = []
+
+    reminder = 1
+    for project in projects:
```
[the remark this answers is not in the document]
Comment (elcct) [reply datahub/investment/project/notification/emails.py]: How could I forget that. Thank you!

File: datahub/investment/project/notification/emails.py [hunk 4]
```diff
@@ -18,17 +21,75 @@ def send_estimated_land_date_reminder(project, adviser, days_left):
...
+        data = {
+            'number': reminder,
+            **get_project_item(project),
+        }
+
+        notification = get_inner_template_content(
+            NotificationInnerTemplate.NotificationType.UPCOMING_ESTIMATED_LAND_DATE,
+        )
+        for key, value in data.items():
+            notification = notification.replace(f'(({key}))', str(value))
```
[the remark this answers is not in the document]
Comment (elcct) [reply datahub/investment/project/notification/emails.py:86]: I think I did it this way before making `notification_type` unique. Amended now. Thank you!

File: datahub/reminder/tasks.py [hunk 5]
```diff
@@ -63,26 +68,42 @@ def generate_estimated_land_date_reminders_for_subscription(subscription, curren
...
-        ):
-            create_reminder(
-                project=project,
-                adviser=subscription.adviser,
-                days_left=days_left,
-                send_email=subscription.email_reminders_enabled,
-                current_date=current_date,
-            )
+
+    fd = now().date().replace(day=1)  # first day of the month
```
[the remark this answers is not in the document]
Comment (elcct) [reply datahub/reminder/tasks.py]: Updated, thank you!

File: datahub/reminder/tasks.py [hunk 6]
```diff
@@ -63,26 +68,42 @@ def generate_estimated_land_date_reminders_for_subscription(subscription, curren
...
+        )
+
+    for project in projects:
+        day_diff = (project.estimated_land_date - current_date).days
+        days_left = 30 if day_diff < 32 else 60
+        create_reminder(
+            project=project,
+            adviser=subscription.adviser,
+            days_left=days_left,
+            # don't sent emails for each project is summary notification threshold has been reached
```
[the remark this answers is not in the document]
Comment (elcct) [reply datahub/reminder/tasks.py]: Nice one. Thank you!

**Review by richmadetech — APPROVED** [review]

**Review by chloehkwong1 — APPROVED** [review]
Are all this codecov/patch warnings okay to ignore? Will give approval anyway as looks good :)

**Pull request merged**